lib/matplotlib/testing/compare.py

```diff
@@ -380,8 +380,8 @@ def compare_images(expected, actual, tol, in_decorator=False):
         raise IOError('Baseline image %r does not exist.' % expected)
     extension = expected.split('.')[-1]
     if extension != 'png':
-        actual = convert(actual, False)
-        expected = convert(expected, True)
+        actual = convert(actual, cache=False)
+        expected = convert(expected, cache=True)
```

Comment:
(not directly related, but this is clearer and I was going through compare_images to try to understand the source of the warning...)